src/services/device-management/iot-device.service.ts

```diff
@@ -1102,4 +1167,149 @@ export class IoTDeviceService {
+        switch (cast.authenticationType) {
+            case AuthenticationType.PASSWORD:
+                cast.mqttpassword ??= this.encryptionHelperService.basicEncrypt(
+                    settings.mqttpassword
+                );
+                cast.mqttusername = settings.mqttusername;
+                break;
+            case AuthenticationType.CERTIFICATE:
+                cast.caCertificate = settings.caCertificate;
+                cast.deviceCertificate = settings.deviceCertificate;
+                cast.deviceCertificateKey = this.encryptionHelperService.basicEncrypt(
+                    settings.deviceCertificateKey
+                );
+                break;
+        }
+
+        cast.mqttURL = settings.mqttURL;
+        cast.mqttPort = settings.mqttPort;
+        cast.mqtttopicname = settings.mqtttopicname;
+
+        return cast;
+    }
+
+    private async enrichMQTTBrokerDevice(iotDevice: IoTDevice) {
+        const device = iotDevice as MQTTBrokerDeviceDTO;
+        device.mqttBrokerSettings = {
+            authenticationType: device.authenticationType,
+            caCertificate: device.caCertificate,
+            deviceCertificate: device.deviceCertificate,
+            deviceCertificateKey: this.encryptionHelperService.basicDecrypt(
+                device.deviceCertificateKey
+            ),
+            mqtttopicname: device.mqtttopicname,
+            mqttURL: device.mqttURL,
+            mqttPort: device.mqttPort,
+            mqttusername: device.mqttusername,
+            mqttpassword: device.mqttpassword,
+            permissions: device.permissions,
+        };
+        return device;
+    }
+
+    private async enrichMQTTSubscriberDevice(iotDevice: IoTDevice) {
+        const device = iotDevice as MQTTSubscriberDeviceDTO;
+        device.mqttSubscriberSettings = {
+            authenticationType: device.authenticationType,
+            caCertificate: device.caCertificate,
+            deviceCertificate: device.deviceCertificate,
+            deviceCertificateKey: this.encryptionHelperService.basicDecrypt(
+                device.deviceCertificateKey
+            ),
+            mqtttopicname: device.mqtttopicname,
+            mqttURL: device.mqttURL,
+            mqttPort: device.mqttPort,
+            mqttusername: device.mqttusername,
+            mqttpassword: this.encryptionHelperService.basicDecrypt(device.mqttpassword),
+            permissions: MQTTPermissionLevel.read,
+        };
+        return device;
+    }
+
+    private async handleNewMQTTDevices(dbIotDevices: IoTDevice[]) {
+        await this.fixMQTTBrokerTopics(dbIotDevices);
+        await this.createNewMQTTClients(dbIotDevices);
+    }
+
+    private async fixMQTTBrokerTopics(dbIotDevices: IoTDevice[]) {
+        const newMQTTBrokers = dbIotDevices.filter(
+            (d: MQTTBrokerDevice) =>
+                d.type === IoTDeviceType.MQTTBroker &&
+                d.mqtttopicname.includes("undefined")
```

Comment:
Maybe also check for NULL here :-)?

Reply:
What part of it, if there are no new devices this will be an empty array